### PR TITLE
Added new structure that is based on KLI framework.

### PR DESCRIPTION
**Pull request by lukaDoric**
[About KLI framework](https://onlinelibrary.wiley.com/doi/full/10.1111/j.1551-6709.2012.01245.x)

### CODE REVIEW

**Comment by lukaDoric** [comment]
There is some undone work related to submission that we will discuss on monday.

**Comment by lukaDoric** [comment]
KnowledgeComponent is named as UnitKnowledgeComponent because the name clashes with the old KC. We can change this name latter.

File: src/Tutor.Core/DomainModel/IUnitRepository.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using Tutor.Core.DomainModel.Units;
+
+namespace Tutor.Core.DomainModel
+{
+    public interface IUnitRepository
+    {
+        Unit GetUnit(int id);
+        List<Unit> GetUnits();
+
+        UnitKnowledgeComponent GetKnowledgeComponent(int id);
```
Comment (Luburic) [inline src/Tutor.Core/DomainModel/IUnitRepository.cs:11]: We should remember to rename this UnitKC to KC when we finalize the refactoring.

File: src/Tutor.Core/DomainModel/InstructionalEvents/Video.cs [hunk 2]
```diff
@@ -1,4 +1,6 @@
-﻿namespace Tutor.Core.ContentModel.LearningObjects
+﻿using Tutor.Core.ContentModel.LearningObjects;
+
+namespace Tutor.Core.DomainModel.InstructionalEvents
 {
     public class Video : LearningObject
```
Comment (Luburic) [inline src/Tutor.Core/DomainModel/InstructionalEvents/Video.cs:5]: The next step is to establish the correct hierarchy.

File: src/Tutor.Infrastructure/Database/TutorContext.cs [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
+using Tutor.Core.DomainModel.AssessmentEvents.Challenges.FulfillmentStrategy.MetricChecker;
+using Tutor.Core.DomainModel.AssessmentEvents.Challenges.FulfillmentStrategy.NameChecker;
+using Tutor.Core.DomainModel.AssessmentEvents.Questions;
+using Tutor.Core.DomainModel.InstructionalEvents;
+using Tutor.Core.DomainModel.Units;
 using Tutor.Core.LearnerModel.Learners;
 using Tutor.Core.ProgressModel.Feedback;
 using Tutor.Core.ProgressModel.Progress;
 using Tutor.Core.ProgressModel.Submissions;
+using KnowledgeComponent = Tutor.Core.DomainModel.KnowledgeComponents.KnowledgeComponent;
```
Comment (Luburic) [inline src/Tutor.Infrastructure/Database/TutorContext.cs:16]: Once we remove the content model we should remember to fix these using to follow the current convention.

**Review by Luburic — COMMENTED** [review]
Looks good, but I wouldn't merge yet.

We should keep this branch alive and fork additional changes from it. Once we complete the transition to the KLI domain we can merge.

The next branch should tackle establishing the appropriate hierarchy. That is, learning objects should be swapped for IEs and AEs accordingly.

We should work towards removing the ContentModel and to enable the Domain Model to satisfy the following use cases:

- Get all units
- Get a single unit with its knowledge components
- Get a knowledge node for the selected knowledge components and load the related IEs and AEs.

After that we can tackle submissions and evaluations.

**Pull request merged**